tests/expr_and_series/arithmetic_test.py

```diff
@@ -157,6 +157,60 @@ def test_truediv_same_dims(
     assert_equal_data({"a": result}, {"a": [2, 1, 1 / 3]})
 
 
+@pytest.mark.parametrize(("left", "right"), [(-2, 0), (0, 0), (2, 0)])
+def test_series_div_by_zero(
+    left: int,
+    right: int,
+    constructor_eager: ConstructorEager,
+    request: pytest.FixtureRequest,
+) -> None:
+    data: dict[str, list[int]] = {"a": [left], "b": [right]}
+    div_by_zero_results: set[float | None] = {float("inf"), float("-inf"), None}
+    df = nw.from_native(constructor_eager(data), eager_only=True)
+    if "pyarrow" in str(constructor_eager):
+        # pyarrow floordiv cannot divide by zero
+        request.applymarker(pytest.mark.xfail)
+        truediv_result = df["a"] / df["b"]  # truediv
+        floordiv_result = df["a"] // df["b"]  # floordiv
+        assert truediv_result[0] in div_by_zero_results or truediv_result.is_nan()[0]
+        assert floordiv_result[0] in div_by_zero_results or floordiv_result.is_nan()[0]
+    truediv_result = df["a"] / df["b"]  # truediv
+    floordiv_result = df["a"] // df["b"]  # floordiv
+    assert truediv_result[0] in div_by_zero_results or truediv_result.is_nan()[0]
+    assert floordiv_result[0] in div_by_zero_results or floordiv_result.is_nan()[0]
```

Comment:
What about having an `expected` parameter together with `left` and `right` so that you can specify the exact expected output for each pair instead of having it generic? Then you can use `assert_data_equal` with manually constructed dicts:

`assert_data_equal({"x": truediv_result}, {"x": [expected]}`